src/Tickets/Event.php

```diff
@@ -36,4 +36,23 @@ public static function get_post_type() {
 		return static::$post_type;
 	}
 
+	/**
+	 * Will filter event ID input to a filtered and relevant ID.
+	 *
+	 * @since TBD
+	 *
+	 * @param numeric $event_id The event ID to be normalized.
+	 *
+	 * @return int|null The filtered value.
+	 */
+	public static function filter_event_id( $event_id ): ?int {
+		/**
+		 * This filter allows retrieval of an event ID to be filtered before being accessed elsewhere.
+		 *
+		 * @since TBD
+		 *
+		 * @param int|null The event ID to be filtered.
+		 */
+		return apply_filters( 'tec_tickets_filter_event_id', $event_id );
```

Comment:
Since filters are the Wild West of "let's see what we can retur here", I suggest casting to `int` or `null` to avoid issues with the return value type-hint:
```suggestion
		return (int)apply_filters( 'tec_tickets_filter_event_id', $event_id ) ?: null;
```